Stimpak.cast feature

warang580 opened the pull request.

Commit:
feat: stimpak.cast() [wip]

Commit:
feat: stimpak.cast done

Commit:
feat: unnecessary import

dcrockwell commented on source/spec/stimpak/stimpak.generate.cast.spec.js:
```diff
@@ -0,0 +1,54 @@
+describe("stimpak.generate() (casted answers)", () => {
+	let stimpak;
+
+	let callbackA = answer => parseInt(answer);
```
Fix whitespacing to be tabs-only

dcrockwell commented on source/spec/stimpak/stimpak.casts.spec.js:
```diff
@@ -0,0 +1,26 @@
+
+describe("stimpak.casts()", () => {
+	let stimpak;
+  let callbackA = answer => parseInt(answer);
```
whitespacing: tabs only

dcrockwell commented:
Very nice, I left a couple of formatting consideration notes

dcrockwell commented:
There's an extra scenario as well which needs a test to cover off on it (note there is no `.prompt`):

instead of this, you can do:

``` javascript
stimpak
.destination(process.cwd())
.answers({
    "casted": "5",
    "untouched": "5"
})
.cast(answer => parseInt(answer))
.generate(error => {
    stimpak.answers().casted.should.eql(5);
    stimpak.answers().untouched.should.eql("5");
});
```

Commit:
feat: check .cast on .answers + fixed tabs

warang580 commented:
I've added the test and cleaned the indentation with tabs only.

Just one question, what is the expected behaviour when doing this : 

``` javascript
stimpak
  .cast(answer => parseInt(answer))
  .prompt(promptOne)
  .prompt(promptTwo)
```

Are promptTwo answers (and also following prompts answers) also casted ? Or is it just the first .prompt() after .cast() ?

dcrockwell commented:
> Just one question, what is the expected behaviour when doing this :

For now, I imagine the .cast working on all subsequent prompts, but in the future we could make it per-prompt
